pkg/reconciler/common/prune.go

```diff
@@ -422,7 +439,9 @@ func (pruner *Pruner) checkAnnotationsRemovedNamespaces(ctx context.Context) ([]
 	return uniqueCronName, err
 }
 
-func diffLastAppliedPruneConfig(ctx context.Context, k kubernetes.Interface, defaultconf v1alpha1.Prune, ns *corev1.Namespace) (bool, error) {
+func diffLastAppliedPruneConfig(ctx context.Context, k kubernetes.Interface, tC *v1alpha1.TektonConfig, ns *corev1.Namespace) (bool, error) {
+	// Pruner Config
+	defaultconf := tC.Spec.Pruner
```

Comment:
I think here we need `tC.Spec.Config` instead of `tC.Spec.Pruner`, as we want to check if there is a change in NodeSelector  or Tolerations

Reply:
we have computed the `tC.Spec.Config` below here I just added a comment so that we do not get confused with Pruner config and there is a difference between them

Reply:
How about creating a different check itself for the tolerations and nodeselectors. 
And to check the last applied config we can set an annotation on the cronjob itself.

Reply:
Why we want to do it only on openshift? we have the feature of providing the node selector and tolerations on config in both openshift and k8s afaik,